weblab/experiments/views.py

```diff
@@ -258,7 +258,7 @@ def get(self, request, *args, **kwargs):
             }
         }
 
-        if len(versions) <= len(pks):
+        if len(versions) < len(pks):
```

Comment:
Good spot. However, testing it here I find it now never shows! I think this is because this check is in the JSON view, which only gets the visible version IDs in its URL, since the main view has already filtered them. So I think we need the main view to fill in the `notes.errors` context variable, and the `weblab/templates/includes/notifications.html` template should display these (at present that loop is commented out).

Reply:
Good point - that never got converted over from the original JSP. I think it's doing the right thing now!